apphub/image_generation/pggan_nihchestxray/pggan_nihchestxray.py

```diff
@@ -154,62 +155,79 @@ def on_epoch_end(self, state):
+                               ops=[resize_scheduler_128, lowres_op, rescale_x, rescale_lowres])
+    pipeline_1024 = fe.Pipeline(batch_size=batchsize_scheduler_1024,
+                                data=writer_1024,
+                                ops=[resize_scheduler_1024, lowres_op, rescale_x, rescale_lowres])
+
+    pipeline_scheduler = Scheduler({0: pipeline_128, 55: pipeline_1024})
 
-    # In Pipeline, we use the schedulers for batch_size and ops.
-    pipeline = fe.Pipeline(
-        batch_size=batchsize_scheduler,
-        data=writer,
-        ops=[
-            resize_scheduler,
-            CreateLowRes(inputs="x", outputs="x_lowres"),
-            Rescale(inputs="x", outputs="x"),
-            Rescale(inputs="x_lowres", outputs="x_lowres")
-        ])
-
-    opt2 = tf.keras.optimizers.Adam(learning_rate=0.001, beta_1=0.0, beta_2=0.99, epsilon=1e-8)
-    opt3 = tf.keras.optimizers.Adam(learning_rate=0.001, beta_1=0.0, beta_2=0.99, epsilon=1e-8)
-    opt4 = tf.keras.optimizers.Adam(learning_rate=0.001, beta_1=0.0, beta_2=0.99, epsilon=1e-8)
-    opt5 = tf.keras.optimizers.Adam(learning_rate=0.001, beta_1=0.0, beta_2=0.99, epsilon=1e-8)
-    opt6 = tf.keras.optimizers.Adam(learning_rate=0.001, beta_1=0.0, beta_2=0.99, epsilon=1e-8)
-    opt7 = tf.keras.optimizers.Adam(learning_rate=0.001, beta_1=0.0, beta_2=0.99, epsilon=1e-8)
+    optimizer = tf.keras.optimizers.Adam(learning_rate=0.001, beta_1=0.0, beta_2=0.99, epsilon=1e-8)
 
     fade_in_alpha = tf.Variable(initial_value=1.0, dtype='float32', trainable=False)
 
-    d2, d3, d4, d5, d6, d7 = fe.build(model_def=lambda: build_D(fade_in_alpha=fade_in_alpha, target_resolution=7, num_channels=1),
-                                      model_name=["d2", "d3", "d4", "d5", "d6", "d7"],
-                                      optimizer=[opt2, opt3, opt4, opt5, opt6, opt7],
-                                      loss_name=["dloss", "dloss", "dloss", "dloss", "dloss", "dloss"])
+    d2, d3, d4, d5, d6, d7, d8, d9, d10 = fe.build(
+        model_def=lambda: build_D(fade_in_alpha=fade_in_alpha, target_resolution=10, num_channels=1),
+        model_name=["d2", "d3", "d4", "d5", "d6", "d7", "d8", "d9", "d10"],
+        optimizer=[optimizer]*9,
+        loss_name=["dloss"]*9)
 
-    g2, g3, g4, g5, g6, g7, G = fe.build(model_def=lambda: build_G(fade_in_alpha=fade_in_alpha, target_resolution=7, num_channels=1),
-                                         model_name=["g2", "g3", "g4", "g5", "g6", "g7", "G"],
-                                         optimizer=[opt2, opt3, opt4, opt5, opt6, opt7, opt7],
-                                         loss_name=["gloss", "gloss", "gloss", "gloss", "gloss", "gloss", "gloss"])
+    g2, g3, g4, g5, g6, g7, g8, g9, g10, G = fe.build(
+        model_def=lambda: build_G(fade_in_alpha=fade_in_alpha, target_resolution=10, num_channels=1),
+        model_name=["g2", "g3", "g4", "g5", "g6", "g7", "g8", "g9", "g10", "G"],
+        optimizer=[optimizer]*10,
```

Comment:
same optimizer for g and d?

Reply:
yep.

Reply:
is that to prevent the discriminator from learning too fast?

Reply:
no, to save some gpu memory.